### PR TITLE
set correct test dependencies

### PR DESCRIPTION
**Pull request by tjhei**
There was a missing dependency from the .cmp.notime file to the
normalization of all (generated and reference) outputs. This caused the
tester to occasionally fail. Fix this.

### CODE REVIEW

**Comment by bangerth** [comment]
No. Nice try, but it doesn't even pass `cmake` :-)

**Comment by tjhei** [comment]
second try :-)

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -257,27 +257,23 @@ FOREACH(_test ${_tests})
...
     ENDIF()
 
-
-
+    # command to normalize all output files and all reference output files
+    ADD_CUSTOM_TARGET(${_test}.normalize
+	COMMAND ${PERL_EXECUTABLE} -pi
+		  ${CMAKE_CURRENT_SOURCE_DIR}/normalize.pl
+		  `find ${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/ -type f`
+	DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/screen-output)
```
Comment (bangerth) [inline tests/CMakeLists.txt]: yes, nice job de-duplicating code

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -307,6 +304,17 @@ FOREACH(_test ${_tests})
...
       )
 
+      # The screen-output target needs to normalize the .cmp.notime
+      # tests. They need to exist first, so add an explicit dependency. To do
+      # that, we first create a target to the command above (ADD_DEPENDENCIES
+      # can only deal with targets, not commands). Finally, we need to remove "/"
+      # (illegal in target names).
+      STRING(REPLACE "/" "-" _target_name_copy_output make-${_test}-${_output}.cmp.notime)
+      ADD_CUSTOM_TARGET(${_target_name_copy_output}
+	${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/${_output}.cmp.notime)
```
Comment (bangerth) [inline tests/CMakeLists.txt]: did you mean to say `DEPENDS` on the second line?

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -318,9 +326,7 @@ FOREACH(_test ${_tests})
 	  | egrep -v '^--'
 	  | sed 's\#${ASPECT_BASE_DIR}\#ASPECT_DIR\#g'
 	  > ${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/${_output}.notime
-	DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/screen-output
-		${CMAKE_CURRENT_BINARY_DIR}/output-${_test}/${_output}.cmp.notime
-      )
+	DEPENDS ${_test}.normalize)
```
Comment (bangerth) [inline tests/CMakeLists.txt]: yes, nice.

Comment (bangerth) [inline tests/CMakeLists.txt]: I think this is the issue here?

Comment (bangerth) [inline tests/CMakeLists.txt]: ? -> !

**Comment by tjhei** [comment]
it looks like it is working?!?

**Comment by bangerth** [comment]
Let's give it a shot!

**Pull request merged**